[ao][sparsity] Refactor and bugfix SparseQuantizedLayer tests

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81803
* __->__ #81802

Summary: Needed to refactor this PR to add tests for some new layers without copy pasting the entirety of the code. Its basically just a helper that does exactly what the other tests did since they were essentially copies of one another. Its possible to do similar with the quantized kernels test but its different enough that it seemed more effort than it was worth. Also bugfix: Originally line 150  I believe was wrong since model.weight is never used, though the only effect was that the specific weight wasn't used.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseLayers

Reviewers:

Subscribers:

Tasks:

Tags:

HDCharles committed:
[ao][sparsity] Refactor and bugfix SparseQuantizedLayer tests

Summary: Needed to refactor this PR to add tests for some new layers without copy pasting the entirety of the code. Its basically just a helper that does exactly what the other tests did since they were essentially copies of one another. Its possible to do similar with the quantized kernels test but its different enough that it seemed more effort than it was worth. Also bugfix: Originally line 150  I believe was wrong since model.weight is never used, though the only effect was that the specific weight wasn't used.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseLayers

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] Refactor and bugfix SparseQuantizedLayer tests"


Summary: Needed to refactor this PR to add tests for some new layers without copy pasting the entirety of the code. Its basically just a helper that does exactly what the other tests did since they were essentially copies of one another. Its possible to do similar with the quantized kernels test but its different enough that it seemed more effort than it was worth. Also bugfix: Originally line 150  I believe was wrong since model.weight is never used, though the only effect was that the specific weight wasn't used.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseLayers

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] Refactor and bugfix SparseQuantizedLayer tests"


Summary: Needed to refactor this PR to add tests for some new layers without copy pasting the entirety of the code. Its basically just a helper that does exactly what the other tests did since they were essentially copies of one another. Its possible to do similar with the quantized kernels test but its different enough that it seemed more effort than it was worth. Also bugfix: Originally line 150  I believe was wrong since model.weight is never used, though the only effect was that the specific weight wasn't used.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseLayers

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] Refactor and bugfix SparseQuantizedLayer tests"


Summary: Needed to refactor this PR to add tests for some new layers without copy pasting the entirety of the code. Its basically just a helper that does exactly what the other tests did since they were essentially copies of one another. Its possible to do similar with the quantized kernels test but its different enough that it seemed more effort than it was worth. Also bugfix: Originally line 150  I believe was wrong since model.weight is never used, though the only effect was that the specific weight wasn't used.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseLayers

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] Refactor and bugfix SparseQuantizedLayer tests"


Summary: Needed to refactor this PR to add tests for some new layers without copy pasting the entirety of the code. Its basically just a helper that does exactly what the other tests did since they were essentially copies of one another. Its possible to do similar with the quantized kernels test but its different enough that it seemed more effort than it was worth. Also bugfix: Originally line 150  I believe was wrong since model.weight is never used, though the only effect was that the specific weight wasn't used.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseLayers

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] Refactor and bugfix SparseQuantizedLayer tests"


Summary: Needed to refactor this PR to add tests for some new layers without copy pasting the entirety of the code. Its basically just a helper that does exactly what the other tests did since they were essentially copies of one another. Its possible to do similar with the quantized kernels test but its different enough that it seemed more effort than it was worth. Also bugfix: Originally line 150  I believe was wrong since model.weight is never used, though the only effect was that the specific weight wasn't used.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseLayers

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] Refactor and bugfix SparseQuantizedLayer tests"


Summary: Needed to refactor this PR to add tests for some new layers without copy pasting the entirety of the code. Its basically just a helper that does exactly what the other tests did since they were essentially copies of one another. Its possible to do similar with the quantized kernels test but its different enough that it seemed more effort than it was worth. Also bugfix: Originally line 150  I believe was wrong since model.weight is never used, though the only effect was that the specific weight wasn't used.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseLayers

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] Refactor and bugfix SparseQuantizedLayer tests"


Summary: Needed to refactor this PR to add tests for some new layers without copy pasting the entirety of the code. Its basically just a helper that does exactly what the other tests did since they were essentially copies of one another. Its possible to do similar with the quantized kernels test but its different enough that it seemed more effort than it was worth. Also bugfix: Originally line 150  I believe was wrong since model.weight is never used, though the only effect was that the specific weight wasn't used.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseLayers

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on test/ao/sparsity/test_kernels.py:
```diff
@@ -110,232 +109,204 @@ def test_sparse_qlinear(self):
+
+        # PREPARE MODELS FOR QUANTIZATION
+        # -------------------------------
+        model.linear.weight = nn.Parameter(W_q.dequantize())
```
so IIUC this step isn't really required, without this the test would just run for some dummy weight in `model.linear`?

HDCharles replied:
yeah, i left this in because it seems necessary if we are to eventually refactor this with the kernel tests but you're right, it doesn't do a ton for the layer based tests.

HDCharles committed:
Update on "[ao][sparsity] Refactor and bugfix SparseQuantizedLayer tests"


Summary: Needed to refactor this PR to add tests for some new layers without copy pasting the entirety of the code. Its basically just a helper that does exactly what the other tests did since they were essentially copies of one another. Its possible to do similar with the quantized kernels test but its different enough that it seemed more effort than it was worth. Also bugfix: Originally line 150  I believe was wrong since model.weight is never used, though the only effect was that the specific weight wasn't used.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseLayers

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented:
@pytorchmergebot merge -g
